ui-v2/app/instance-initializers/selection.js

```diff
@@ -0,0 +1,49 @@
+const isSelecting = function(win = window) {
+  const selection = win.getSelection();
+  let selecting = false;
+  try {
+    selecting = 'isCollapsed' in selection && !selection.isCollapsed;
+  } catch (e) {
+    // passthrough
+  }
+  return selecting;
+};
+export default {
+  name: 'selection',
+  initialize(container) {
+    const dom = container.lookup('service:dom');
+    const findAnchor = function(el) {
+      return el.tagName === 'A' ? el : dom.closest('a', el);
+    };
+    const mousedown = function(e) {
+      const $a = findAnchor(e.target);
+      if ($a) {
+        const href = $a.getAttribute('href');
+        if (href) {
+          $a.dataset.href = href;
+          $a.removeAttribute('href');
+        }
+      }
+    };
+    const mouseup = function(e) {
+      const $a = findAnchor(e.target);
```

Comment:
> There's no guaranteeing that the nearest anchor on the up target will be the same as the nearest anchor on the down target.

Nope you're right, there's no guaranteeing. But I don't think that's a problem.

I thought about this when I did it, and I 'think' I've made it so the `mouseup` is the important thing here. i.e. if you haven't `mousedown`ed on an element, then the `href` won't have been swapped out and it will still navigate you away from the page. So it will work as it does now, navigate away, not 100% ideal, but it won't 'break' as such. The swapped out `href` will eventually get swapped back in again on a `mouseup` if you hit the case the other way around (you've somehow `mousedown`ed a link without 'mousingup' on it).

All in all I think this is unlikely to be a problem, it's pretty edge casey, and the gains here are probably worth it?

Reply:
Yeah, I agree it's edge casey. Another way this could go 💥 would be this sequence of events:

1. Mousedown on anchor X
2. Mouseup on anchor Y
3. Keyboard navigation to anchor X
4. Enter key to follow anchor which no longer has an href

Now that I've given this some more thought, what's stopping a simple `e.preventDefault` from working here?

Reply:
Ah ha, good one! Edge casey again but a possibility.

> Now that I've given this some more thought, what's stopping a simple e.preventDefault from working here?

🤔 Oh yeah, maybe! 😂 Classic! Not sure if I'll get chance to try this today, but you're right I can't think of why that wouldn't work. Let's see where I get.